src/Logic.hs

```diff
@@ -486,9 +485,10 @@ tryIntegratePullRequest pr state =
   in do
     result <- tryIntegrate mergeMessage candidate $ Pr.alwaysAddMergeCommit approvalType
     case result of
-      Left (IntegrationFailure targetBranch) ->
+      Left (IntegrationFailure (targetBranch, reason)) -> do
         -- If integrating failed, perform no further actions but do set the
         -- state to conflicted.
+        when (reason == WrongFixups) $ leaveComment pr "PR cannot be merged since it contains fixup commits that do not belong to any other commits."
```

Comment:
I think this is not the right place for leaving this comment, as hoff will then continue with the code below, where it sets the integration status and marks the PR as needing feedback.

This will cause it to eventually reach
https://github.com/channable/hoff/blob/20c6e6708bf43219bab560aedd32ed3009f7dca0/src/Logic.hs#L632-L634
which will call
https://github.com/channable/hoff/blob/20c6e6708bf43219bab560aedd32ed3009f7dca0/src/Logic.hs#L586-L588
for generating the message to leave.

For the "Conflicted" state, it will currently always leave a comment about failed rebasing, which is misleading.

The better approach here would be to include the `reason` in the `Conflicted` state of the PR, then handle that case in [`classifyPullRequest`](https://github.com/channable/hoff/blob/20c6e6708bf43219bab560aedd32ed3009f7dca0/src/Project.hs#L273-L283) by adding a new kind of status to:

https://github.com/channable/hoff/blob/20c6e6708bf43219bab560aedd32ed3009f7dca0/src/Project.hs#L89-L97